### DISCUSSION TITLE
package routes

### DISCUSSION THREAD

**Discussion by ioucyf** [sveltejs/kit#6661] [Ideas]
routes folder is quickly building up and starting to get very cluttered. 

It would be nice to have the ability to _package_ a group of routes and be able to import them into a SvelteKit project like any other npm package. managing different group of routes as individual parts of the projects each with it's own repo etc. 

concept example using express:

An express.js sub project (package) for user-auth:

1. `npm i express` 
2. `import { Router } from 'express';`
3. define routes + methods for:  

- `router('/login', (req, res) => {})` 
- `router('/logout', (req, res) => {})` 
- `router('/signup', (req, res)=> {})`

4. `export default router;`
5. commit + publish package in npm or github registery

An express.js main project (app) would then:
1. `npm i user-auth`
2. `import auth from 'user-auth';`
3. `express.use('/auth', auth);`

Now think SvelteKit instead of express.

it may make more sense to be part of the **Package API of SvelteKit**? 

_For example: if Svelte package would compile to a component library, then SvelteKit package would compile to a route library._

Or maybe not...

Either way, I think this will make it a lot easier to manage SvelteKit projects and even have more options provided by the open source community.

**Comment by j4w8n** [comment]
I don't really know what I'm talking about, but it seems like you can do this now by creating your routing package, importing it in `handle()`, and feed it requests?

> handle
This function runs every time the SvelteKit server receives a [request](https://kit.svelte.dev/docs/web-standards#fetch-apis-request) — whether that happens while the app is running, or during [prerendering](https://kit.svelte.dev/docs/page-options#prerender) — and determines the [response](https://kit.svelte.dev/docs/web-standards#fetch-apis-response). It receives an event object representing the request and a function called resolve, which renders the route and generates a Response. This allows you to modify response headers or bodies, or **bypass SvelteKit entirely (for implementing routes programmatically, for example)**.
